packages/components/src/components/hds/stepper/nav/index.ts

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { schedule } from '@ember/runloop';
+import { assert } from '@ember/debug';
+import { modifier } from 'ember-modifier';
+import type { ComponentLike } from '@glint/template';
+
+import type { HdsStepperNavStepSignature } from './step';
+import type { HdsStepperNavPanelSignature } from './panel';
+import { HdsStepperTitleTagValues } from '../types.ts';
+import type {
+  HdsStepperTitleTags,
+  HdsStepperNavStepIds,
+  HdsStepperNavStep,
+  HdsStepperNavPanelIds,
+} from '../types.ts';
+
+export interface HdsStepperNavSignature {
+  Args: {
+    steps?: HdsStepperNavStep[];
+    currentStep?: number;
+    isInteractive?: boolean;
+    titleTag?: HdsStepperTitleTags;
+    onStepChange?: (event: MouseEvent, stepNumber: number) => void;
+  };
+  Blocks: {
+    body?: [];
+    default: [
+      {
+        Step?: ComponentLike<HdsStepperNavStepSignature>;
+        Panel?: ComponentLike<HdsStepperNavPanelSignature>;
+      },
+    ];
+  };
+  Element: HTMLElement;
+}
+
+export default class HdsStepperNav extends Component<HdsStepperNavSignature> {
+  @tracked private _stepIds: HdsStepperNavStepIds = [];
+  @tracked private _stepNodes: HTMLElement[] = [];
+  @tracked private _panelNodes: HTMLElement[] = [];
+  @tracked private _panelIds: HdsStepperNavPanelIds = [];
+
+  private _setUpStepperNav = modifier(() => {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      assert(
+        'The number of Steps must be equal to the number of Panels',
+        this._stepNodes.length === this._panelNodes.length
+      );
+    });
+
+    return () => {};
+  });
+
+  /**
+   * @param currentStep
+   * @type {number}
+   * @default 0
+   */
```

Comment:
You could delete comments like these. We don't use them anymore. If the function of something needs explanation to clarify just add a simple comment. Most of these seem pretty self explanatory though.

Delete:
```
  /**
   * @param currentStep
   * @type {number}
   * @default 0
   */
```